src/tiledb/cloud/utilities/_common.py

```diff
@@ -200,3 +202,44 @@ def max_memory_usage() -> int:
         result = 0
 
     return result
+
+
+def process_stream(
+    uri: str,
+    cmd: str,
+    read_size: int = 16 << 20,
+) -> Tuple[str, str]:
+    """
+    Process a stream of data from VFS with a subprocess.
+
+    If the file is large and the subprocess only reads a small amount of data,
+    then reduce `read_size` to improve performance.
+
+    :param uri: file URI
+    :param cmd: command to run in the subprocess
+    :param read_size: number of bytes to read per iteration, defaults to 16 MiB
+    :return: stdout and stderr from the subprocess
+    """
+
+    with tiledb.VFS().open(uri) as fp:
+        with subprocess.Popen(
+            cmd.split(),
```

Comment:
To avoid problems caused by e.g. spaces in filenames or other unexpected behavior, I would recommend passing this in as the list of arguments to be passed to the process.

Reply:
Good point, changing that.